docs/lang/articles/basic/sparse.md

```diff
@@ -170,31 +172,52 @@ The bitmasked SNodes are like dense SNodes with auxiliary activity values.
 
 ### Dynamic SNode
 
-To support variable-length fields, Taichi provides dynamic SNodes. The code snippet below first creates a 5x1 dense block (line 2). Then each cell of the dense block contains a variable-length dynamic container (line 3). The maximum length of the dynamic container is 5. In the `make_lists()` function, you can use `ti.append()` to add a value to the end of a dynamic SNode. `x.parent()` is the same as `pixel`. The dense field `l` stores the length of each dynamic SNode.
+To support variable-length fields, Taichi provides dynamic SNodes.
 
-```python {3} title=dynamic.py
-x = ti.field(ti.i32)
-block = ti.root.dense(ti.i, 5)
-pixel = block.dynamic(ti.j, 5)
-pixel.place(x)
-l = ti.field(ti.i32)
-ti.root.dense(ti.i, 5).place(l)
-
-@ti.kernel
-def make_lists():
-    for i in range(5):
-        for j in range(i):
-            ti.append(x.parent(), i, j * j)  # ti.append(pixel, i, j * j)
-        l[i] = ti.length(x.parent(), i)  # [0, 1, 2, 3, 4]
-```
+The first argument of `dynamic` is the axis, the second argument is the maximum length,
+and the third argument (optional) is the `chunk_size`.
 
+The `chunk_size` specifies how much space should the dynamic SNode allocate when the space already allocated runs out.
+For example, if `chunk_size=4`, the dynamic SNode initially allocates the space for 4 elements.
+When the 5th (, 9th, 13th...) element is appended, the dynamic SNode further allocates the space for another 4 elements.
 
 <center>
 
 ![Dynamic](https://raw.githubusercontent.com/taichi-dev/public_files/master/taichi/doc/dynamic.png)
```

Comment:
It seems that the picture no longer matches the code snippet below. Shall we make a new one?